components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5600,16 +5617,51 @@ private boolean isMultivaluedAttribute(String name) {
         }
     }
 
+    private SCIMDefinitions.DataType getCustomAttrDataType(String dataType) {
+
+        if (SCIMDefinitions.DataType.BOOLEAN.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.BOOLEAN;
+        } else if (SCIMDefinitions.DataType.DECIMAL.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.DECIMAL;
+        } else if (SCIMDefinitions.DataType.INTEGER.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.INTEGER;
+        } else if (SCIMDefinitions.DataType.DATE_TIME.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.DATE_TIME;
+        } else if (SCIMDefinitions.DataType.BINARY.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.BINARY;
+        } else if (SCIMDefinitions.DataType.REFERENCE.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.REFERENCE;
+        } else if (SCIMDefinitions.DataType.COMPLEX.name().equalsIgnoreCase(dataType)) {
+            return SCIMDefinitions.DataType.COMPLEX;
```

Comment:
Shall we use switch case here instead of multiple if else of blocks

Reply:
Then we have to convert both strings to lowercase first and use switch case as we need to compare them using equalsIgnoreCase() method here.